tools/shared/src/main/scala/org/scalajs/core/tools/sourcemap/SourceMapWriter.scala

```diff
@@ -100,10 +143,10 @@ class SourceMapWriter(
   }
 
   private def writeHeader(): Unit = {
-    out.write("{\n")
-    out.write("\"version\": 3,\n")
-    out.write("\"file\": " + jsonString(generatedFile) + ",\n")
-    out.write("\"mappings\": \"")
+    val out1 = out
```

Comment:
If it does help, it should however be in a separate commit. It's current owning commit says "reduce number of allocations".